README.md

```diff
@@ -1 +1,156 @@
+[![Build Status](https://travis-ci.com/AccelByte/common-blob-go.svg?branch=master)](https://travis-ci.com/AccelByte/common-blob-go)
+
 # common-blob-go
+Go library to work with AWS(amazon web services) S3 and GCP(google cloud platform) cloud storage
+
+## Usage
+
+### Install
+
+```
+go get -u github.com/AccelByte/common-blob-go
+```
+
+### Importing
+
+```go
+commonblobgo "github.com/AccelByte/common-blob-go"
+```
+
+To create a new storage client, use this function:
+
+```go
+storage, err := storage, err := NewCloudStorage(
+    ctx,
+    isTesting,
+    bucketProvider,
+    bucketName,
+    awsS3Endpoint,
+    awsS3Region,
+    awsS3AccessKeyID,
+    awsS3SecretAccessKey,
+    gcpCredentialsJSON,
+    gcpStorageEmulatorHost,
+)
+```
+
+``NewCloudStorage`` requires such parameters :
+ * ctx context.Context : a context that could be cancelled to force-stop the initialization
+ * isTesting bool : a flag to switch between external and in-docker-compose dependencies. Used from tests
+ * bucketProvider string : provider type. Could be `aws` or `gcp`
+ * bucketName string : the name of a bucket
+
+ * awsS3Endpoint string : S3 endpoint. Used only from tests(required if bucketProvider==`aws` and isTesting == `true`)
+ * awsS3Region string : S3 region(required if bucketProvider==`aws`)
+ * awsS3AccessKeyID string : S3 Access key(required if bucketProvider==`aws`)
```

Comment:
do we support deployment without AWS key? for example by adding AWS role to the ec2.

Reply:
@thoriqsatriya currently, we don't support other methods of authentification, except security keys. I'll add it in the next PR